cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -205,6 +205,11 @@ func run(_ <-chan struct{}) {
 					continue
 				}
 
+				if err := cleanUnschedulable(nodes, kubeClient, recorder); err != nil {
```

Comment:
Why do you need to clean unschedulable nodes here? My feeling is that you only mark them unschedulable right before removing them.